### PR TITLE
Implementing previousTeam attribute of entity

### LINKED ISSUE #26: [Feature Request] - Create previousTeam attribute on Judge entity
### Pre-requisites:
- [x] I looked through both [open and closed issues](../issues?utf8=✓&q=is%3Aissue) and did not find another request for the same or similar feature.

## Describe the solution you'd like

Due to complexity with skipping teams and potentially delaying visits in the future, we should update the `judge` entity to have a new `previousTeam` attribute that will have the same type as `currentTeam`. When a judge skips a team, the `currentTeam` should be added to `visitedTeams` but the `previousTeam` entity should remain the same. When a team is judged normally, the `currentTeam` should be added to `visitedTeams` **and** moved to `previousTeam`.

### PR DESCRIPTION
**Pull request by SpencerKaiser**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../THIRD-PARTY-NOTICES.txt)

## Related Issues

Closes #26 
Fixing an issue where Jest `--watch` was not working appropriately

### CODE REVIEW

File: src/entities/judge.ts [hunk 1]
```diff
@@ -20,23 +20,23 @@ export class Judge extends BaseEntity {
...
-  }
-
   async continue(): Promise<void> {
     await this.recordCurrentTeamAndSave();
   }
 
   async skip(): Promise<void> {
-    await this.clearCurrentTeamAndSave();
+    const updatePrevious = false;
+    await this.recordCurrentTeamAndSave(updatePrevious);
```
Comment (johnkahn) [inline src/entities/judge.ts]: Is this just to make it clear why it's `false`?

**Comment by SpencerKaiser** [comment]
@Johnkahn whoops... definitely forgot the migration

File: src/entities/judge.ts (hunk 1 above)
Comment (SpencerKaiser) [reply src/entities/judge.ts]: Yea that’s exactly why. I figured without it, it just looks like an arbitrary value being passed in and I didn’t want to make the function name super long either.

**Comment by SpencerKaiser** [comment]
@johnkahn just added the migration (which looks like it also picked up the `syncHash` field from one of the previous changes). This is ready for review now 👌

**Pull request merged**